awesome-cis/take_break
# (Base #37) create agent using isomorphic-unfetch

## Description
- `isomorphic-unfetch` 추가 및 Authorization token 추가

### TODO
- 테스트용 console.log 제거
- index.tsx rollback
- 환경에 따라 API_ENDPOINT 처리 추가
- get token from sessionStore

## Related Issue
#30 
## How Has This Been Tested?

1. start server, client, mysql

2. access localhost:3000/

3. check console.log

## Screenshots

browser console
![image](https://user-images.githubusercontent.com/8221797/59362017-bf2f2380-8d6d-11e9-808e-ab942fd4e0c4.png)

server terminal
```
req.headers.authorization token test-authorization-token
Executing (default): SELECT `id`, `isDeleted`, `createdAt`, `updatedAt`, `name`, `email`, `password`, `provider`, `oAuthId` FROM `users` AS `r` WHERE `r`.`email` = 'oyeahhh@naver.com' LIMIT 1;
Executing (default): INSERT INTO `users` (`id`,`createdAt`,`updatedAt`,`name`,`email`,`password`) VALUES (DEFAULT,?,?,?,?,?);
POST /auth/register 200 363.513 ms - 190
```

client terminal
```
data { user: { id: 1, name: 'ryu', email: 'oyeahhh@naver.com' },
  accessToken:
   'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpZCI6MSwiaWF0IjoxNTYwMzUxMjc3fQ.HhQGLQ-t6o-VDfVCT-fF2i8zXVYOLeG39ha2dUZce70' }
props { resData:
   { user: { id: 1, name: 'ryu', email: 'oyeahhh@naver.com' },
     accessToken:
      'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpZCI6MSwiaWF0IjoxNTYwMzUxMjc3fQ.HhQGLQ-t6o-VDfVCT-fF2i8zXVYOLeG39ha2dUZce70' } }
```

mysql terminal
```
mysql> select * from users;
+----+-----------+---------------------+---------------------+------+-------------------+--------------------------------------------------------------+----------+---------+----------+--------------+--------------+------+
| id | isDeleted | createdAt           | updatedAt           | name | email             | password                                                     | provider | oAuthId | nickname | introduction | profileImage | slug |
+----+-----------+---------------------+---------------------+------+-------------------+--------------------------------------------------------------+----------+---------+----------+--------------+--------------+------+
|  1 |      NULL | 2019-06-12 14:54:37 | 2019-06-12 14:54:37 | ryu  | oyeahhh@naver.com | $2b$12$.u/lI8UxD5N5vGvEyEcaM.c0PISqghwpSOtiDT6Xigl6bhSNc08VW | NULL     | NULL    | NULL     | NULL         | NULL         | NULL |
+----+-----------+---------------------+---------------------+------+-------------------+--------------------------------------------------------------+----------+---------+----------+--------------+--------------+------+
1 row in set (0.00 sec)```
